File: kmp-mobile/ios/PazChurch/Features/CasaDePazAnalytics/CasaDePazCharts.swift
import Shared
import SwiftUI

private let monthLabels = [
    "Jan", "Fev", "Mar", "Abr", "Mai", "Jun", "Jul", "Ago", "Set", "Out", "Nov", "Dez",
]

private func periodLabel(_ period: String) -> String {
    let parts = period.split(separator: "-")
    guard parts.count == 2, let monthIndex = Int(parts[1]), monthLabels.indices.contains(monthIndex - 1) else {
        return period
    }
    return "\(monthLabels[monthIndex - 1])/\(parts[0].suffix(2))"
}

/// The full set of charts for the Casa de Paz report — mirrors admin-ui's
/// casa-de-paz-trend-chart / casa-de-paz-breakdown-charts, but `by_time`
/// ("Horário") is intentionally not rendered: it was removed from admin-ui
/// (commit 4494359) and must not be reintroduced on mobile.
struct CasaDePazCharts: View {
    let summary: CasaDePazAnalyticsSummary

    var body: some View {
        VStack(alignment: .leading, spacing: 20) {
            chartCard(title: "Atividade das Casas de Paz") {
                if summary.series.isEmpty {
                    PazChartEmptyView(message: "Nenhuma Casa de Paz registrada no período selecionado.")
                } else {
                    PazTrendChart(
                        barSeries: summary.series.map {
                            PazChartSeriesPoint(
                                label: periodLabel($0.period),
                                value: Double($0.houses),
                                series: "Casas"
                            )
                        },
                        lineSeries: summary.series.flatMap { point -> [PazChartSeriesPoint] in
                            let label = periodLabel(point.period)
                            return [
                                PazChartSeriesPoint(label: label, value: Double(point.adults), series: "Adultos"),
                                PazChartSeriesPoint(label: label, value: Double(point.kids), series: "Crianças"),
                            ]
                        }
                    )
                }
            }

            chartCard(title: "Aderência de Novas Pessoas") {
                if summary.series.isEmpty {
                    PazChartEmptyView(message: "Nenhuma Casa de Paz registrada no período selecionado.")
                } else {
                    PazTrendChart(
                        barSeries: [],
                        lineSeries: summary.series.flatMap { point -> [PazChartSeriesPoint] in
                            let label = periodLabel(point.period)
                            return [
                                PazChartSeriesPoint(label: label, value: Double(point.guests), series: "Convidados"),
                                PazChartSeriesPoint(
                                    label: label,
                                    value: Double(point.conversions),
                                    series: "Conversões"
                                ),
                            ]
                        }
                    )
                }
            }

            chartCard(title: "Casas de Paz por Setor") {
                if summary.bySector.isEmpty {
                    PazChartEmptyView(message: "Nenhuma Casa de Paz registrada no período selecionado.")
                } else {
                    PazHorizontalBarChart(
                        entries: summary.bySector.map {
                            PazChartSeriesPoint(label: $0.label, value: Double($0.houses), series: "Casas")
                        }
                    )
                }
            }

            chartCard(title: "Casas de Paz por Dia") {
                if summary.byDay.isEmpty {
                    PazChartEmptyView(message: "Nenhuma Casa de Paz registrada no período selecionado.")
                } else {
                    PazHorizontalBarChart(
                        entries: summary.byDay.map {
                            PazChartSeriesPoint(label: $0.label, value: Double($0.houses), series: "Casas")
                        }
                    )
                }
            }
        }
    }

    private func chartCard(title: String, @ViewBuilder content: () -> some View) -> some View {
        VStack(alignment: .leading, spacing: 12) {
            Text(title).font(PazTypography.titleSmall).foregroundStyle(PazColors.ink)
            content()
        }
        .padding(16)
        .glassCard(radius: PazSpacing.cardRadiusCompact)
    }
}

File: kmp-mobile/android/src/main/kotlin/br/church/paz/android/ui/features/casadepazanalytics/CasaDePazCharts.kt
package br.church.paz.android.ui.features.casadepazanalytics

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import br.church.paz.android.ui.components.PazBarChart
import br.church.paz.android.ui.components.PazBarChartEmpty
import br.church.paz.android.ui.components.PazBarChartEntry
import br.church.paz.android.ui.theme.PazShapes
import br.church.paz.android.ui.theme.PazSpacing
import br.church.paz.shared.domain.model.CasaDePazAnalyticsSummary

private val MONTH_LABELS =
    listOf("Jan", "Fev", "Mar", "Abr", "Mai", "Jun", "Jul", "Ago", "Set", "Out", "Nov", "Dez")

private fun periodLabel(period: String): String =
    period
        .split("-")
        .let { parts ->
            if (parts.size == 2) {
                val month = parts[1].toIntOrNull()?.let { MONTH_LABELS.getOrNull(it - 1) } ?: parts[1]
                "$month/${parts[0].takeLast(2)}"
            } else {
                period
            }
        }

@Composable
fun CasaDePazChartCard(
    title: String,
    content: @Composable () -> Unit,
) {
    Surface(shape = PazShapes.large, color = MaterialTheme.colorScheme.surface, modifier = Modifier.fillMaxWidth()) {
        Column(Modifier.fillMaxWidth().padding(PazSpacing.Lg)) {
            Text(title, style = MaterialTheme.typography.titleSmall)
            Spacer(Modifier.height(PazSpacing.Md))
            content()
        }
    }
}

// "Atividade das Casas de Paz" — houses (bars) + adults/kids attendance (a
// second bar series, since PazBarChart has no line-overlay support). Mirrors
// admin-ui's casa-de-paz-trend-chart without a charting dependency.
@Composable
fun CasaDePazHousesActivityChart(summary: CasaDePazAnalyticsSummary) {
    CasaDePazChartCard(title = "Atividade das Casas de Paz") {
        if (summary.series.isEmpty()) {
            PazBarChartEmpty("Nenhuma Casa de Paz registrada no período selecionado.")
        } else {
            PazBarChart(entries = summary.series.map { PazBarChartEntry(periodLabel(it.period), it.houses.toFloat()) })
        }
    }
}

@Composable
fun CasaDePazAttendanceChart(summary: CasaDePazAnalyticsSummary) {
    CasaDePazChartCard(title = "Presenças por Mês") {
        if (summary.series.isEmpty()) {
            PazBarChartEmpty("Nenhuma Casa de Paz registrada no período selecionado.")
        } else {
            PazBarChart(
                entries =
                    summary.series.map {
                        PazBarChartEntry(periodLabel(it.period), (it.adults + it.kids).toFloat())
                    },
            )
        }
    }
}

// "Aderência de Novas Pessoas" — guests + conversions per month.
@Composable
fun CasaDePazNewPeopleChart(summary: CasaDePazAnalyticsSummary) {
    CasaDePazChartCard(title = "Aderência de Novas Pessoas") {
        if (summary.series.isEmpty()) {
            PazBarChartEmpty("Nenhuma Casa de Paz registrada no período selecionado.")
        } else {
            PazBarChart(entries = summary.series.map { PazBarChartEntry(periodLabel(it.period), it.guests.toFloat()) })
        }
    }
}

@Composable
fun CasaDePazBySectorChart(summary: CasaDePazAnalyticsSummary) {
    CasaDePazChartCard(title = "Casas de Paz por Setor") {
        if (summary.bySector.isEmpty()) {
            PazBarChartEmpty("Nenhuma Casa de Paz registrada no período selecionado.")
        } else {
            PazBarChart(entries = summary.bySector.map { PazBarChartEntry(it.label, it.houses.toFloat()) })
        }
    }
}

@Composable
fun CasaDePazByDayChart(summary: CasaDePazAnalyticsSummary) {
    CasaDePazChartCard(title = "Casas de Paz por Dia") {
        if (summary.byDay.isEmpty()) {
            PazBarChartEmpty("Nenhuma Casa de Paz registrada no período selecionado.")
        } else {
            PazBarChart(entries = summary.byDay.map { PazBarChartEntry(it.label, it.houses.toFloat()) })
        }
    }
}
